src/hooks.ts

```diff
@@ -208,19 +216,26 @@ export const useDrag = ({ onStart, onMove, onEnd, containerRef, knobs }: UseDrag
   React.useLayoutEffect(() => {
     if (isTouchDevice) {
       const container = containerRef.current
-      container?.addEventListener('touchstart', onTouchStart, { capture: true, passive: false })
-      // we are adding this touchmove listener to cancel drag if user is scrolling
-      // however, it's also important to have a touchmove listener always set
-      // with non-capture and non-passive option to prevent an issue on Safari
-      // with e.preventDefault (https://github.com/atlassian/react-beautiful-dnd/issues/1374)
-      document.addEventListener('touchmove', touchScrollListener, {
-        capture: false,
-        passive: false,
-      })
-      document.addEventListener('touchend', touchScrollListener, {
-        capture: false,
-        passive: false,
-      })
+
+      if (allowDrag) {
+        container?.addEventListener('touchstart', onTouchStart, { capture: true, passive: false })
+        // we are adding this touchmove listener to cancel drag if user is scrolling
+        // however, it's also important to have a touchmove listener always set
+        // with non-capture and non-passive option to prevent an issue on Safari
+        // with e.preventDefault (https://github.com/atlassian/react-beautiful-dnd/issues/1374)
+        document.addEventListener('touchmove', touchScrollListener, {
+          capture: false,
+          passive: false,
+        })
+        document.addEventListener('touchend', touchScrollListener, {
+          capture: false,
+          passive: false,
+        })
+      } else {
```

Comment:
I dont think we need the else part. Even if the allowDrag flips from true to false, the cleanup function will take care of removing the events .

Reply:
I tested state changes in `allowDrag` and it wasn't working w/o this else clause

It could possible be because the cleanup function doesn't pass options like: `{ capture: false }`

Reply:
Oh indeed that could explain it. Could you please try passing the same options to the remove calls so that the cleanup works on its own?

Reply:
Yep that worked, updated the PR